controllers/component/component_adapter.go

```diff
@@ -105,6 +105,11 @@ func (a *Adapter) EnsureComponentIsCleanedUp() (controller.OperationResult, erro
 			return controller.RequeueWithError(err)
 		}
 	}
+	a.component, err = a.loader.GetComponent(a.client, a.context, a.component.Name, a.component.Namespace)
+	if err != nil {
+		a.logger.Error(err, "Failed to get component.")
+		return controller.RequeueWithError(err)
```

Comment:
Might be a good idea to continue processing if the error is `NotFound` - see [here](https://github.com/redhat-appstudio/integration-service/blob/a28863928205fed45ddf7d952466dd9a70387df3/controllers/buildpipeline/buildpipeline_adapter.go#L74-L80) for example.

Reply:
Good idea, updated.

Reply:
Hmm I cutoff the remove finalizer with this logic, I need to move that removal probably in the if condition.